packages/inference/src/InferenceClient.ts

```diff
@@ -34,40 +32,33 @@ export class InferenceClient {
 				enumerable: false,
 				value: (params: RequestArgs, options: Options) =>
 					// eslint-disable-next-line @typescript-eslint/no-explicit-any
-					fn({ ...params, accessToken } as any, { ...defaultOptions, ...options }),
+					fn({ endpointUrl: defaultOptions.endpointUrl, accessToken, ...params } as any, {
+						...defaultOptions,
```

Comment:
should probably remove `endpointUrl` here though

```suggestion
						...omit(defaultOptions, ["endpointUrl"]),
```